analysis/quantifier-comparisons/13-spatial-transcriptomics-benchmarking.Rmd

```diff
@@ -0,0 +1,411 @@
+
+# sample name
+sample_id <- c("SCPCR000372")
+
+# create directories
+if(!dir.exists(data_dir)){
+  dir.create(data_dir, recursive = TRUE)
+}
+
+if(!dir.exists(results_dir)){
+  dir.create(results_dir, recursive = TRUE)
+}
+```
+
+
+```{r}
+mito_file <- file.path(base_dir, "sample-info", "Homo_sapiens.GRCh38.103.mitogenes.txt")
+  
+# read in mito genes 
+mito_genes <- readr::read_tsv(mito_file, col_names = "gene_id")
+mito_genes <- mito_genes %>%
+  dplyr::pull(gene_id) %>%
+  unique()
+```
+
+
+## Alevin-fry + Spaceranger versus Spaceranger Only
+
+Now let's take a look at comparing the two methods of using Alevin-fry + Spaceranger to only Spaceranger for quantification. 
+To do this, we will read in the Alevin-fry + Spaceranger combined and Spaceranger only `SpatialExperiment` objects separately and then merge them into one list before grabbing the per cell and per gene quality metrics. 
+
+### Create Spatial Experiments 
+
+```{r}
+# get path to fry output directory 
+fry_dir <- file.path(s3_out, "alevin-fry-knee", sample_id)
+fry_dir <- paste0(fry_dir, "-spliced_intron_txome_k31-salign-cr-like-em-knee")
+
+# paths to spatial folders 
+cellranger_folder <- paste0(sample_id, "-cdna-spatial")
+spatial_dir <- file.path(s3_out, "cellranger", cellranger_folder, "outs", "spatial")
+```
+
+
+```{r}
+# read in combined fry and spaceranger spe 
+fry_spe <- create_fry_spaceranger_spe(fry_dir, 
+                                       spatial_dir, 
+                                       sample_id)
+```
+
+```{r}
+# spaceranger output paths
+spaceranger_out <- file.path(s3_out, "cellranger", cellranger_folder, "outs")
+
+# read in spaceranger output directly using read10XVisium
+spaceranger_spe <- create_spaceranger_spe(spaceranger_out, sample_id)
+```
+
+### Per Spot QC Metrics
+
+Now that we have read in the data and created our two `SpatialExperiment` objects, we can go ahead and combine them into one list and then calculate the per spot QC metrics using `scuttle::addPerCellQCMetrics().`
+
+```{r}
+# create one list with both spe's together
+all_spe_list <- list(fry_spe, spaceranger_spe)
+names(all_spe_list) <- c("alevin-fry", "spaceranger")
+
+# calculate per cell QC and output to a combined data frame with plotting 
+all_spe_list <- all_spe_list %>%
+    purrr::map(
+      ~ scuttle::addPerCellQCMetrics(.x, 
+                                     subsets = list(mito = mito_genes[mito_genes %in% rownames(.x)]))) %>%
+  purrr::map(scuttle::addPerFeatureQCMetrics)
+```
+
+After adding in the per spot QC metrics to both of the spe's, we want to extract the `colData` from each spe and create a data frame that we can use for plotting. 
+We will also need some information about each sample and how it was run, so we will create a sample metadata table, `sample_info_df` that will then be merged with the `colData`. 
+
+```{r}
+# create sample info dataframe to be joined with per spot dataframe later
+sample_info_df <- quant_info_table(data_dir= s3_out, 
```

Comment:
For clarity here, it might be worth printing this table. It is unclear what the contents will be within the notebook, as the function is elsewhere.